miso-web/src/main/webapp/scripts/workflow_display.js

```diff
@@ -0,0 +1,156 @@
+var WorkflowDisplay = (function() {
+  var display;
+  var loadingTag = jQuery("<img src='/styles/images/ajax-loader.gif'>");
+  var errorDiv = jQuery("<div id='inputError'></div>");
+
+  function ajax(method, url, onSuccess, onError) {
+    showLoading();
+    jQuery.ajax({
+      "type": method,
+      "url": url,
+      "contentType": "application/json; charset=utf8",
+      "dataType": "json",
+      "success": onSuccess,
+      "error": onError
+    })
+  }
+  
+  function showSuccess() {
+    display.empty().append(jQuery("<p>Workflow was successfully completed.</p>"));
+  }
+
+  function executeWorkflow(workflowId) {
+    ajax("POST", "/miso/rest/workflow/" + workflowId + "/execute", showSuccess, showError);
+  }
+
+  function showError(xhr) {
+    loadingTag.remove();
+    display.children().show();
+    display.children("input").focus();
+    errorDiv.text(parseError(xhr));
+  }
+
+  function parseError(xhr) {
+    var responseObj = JSON.parse(xhr.responseText);
```

Comment:
This should get wrapped in a `try` block. The response might be HTML junk if the session has closed.